Minimal utoipa example with file upload

mattdavis90 opened the pull request:
As per #82 I've added a minimal example of using `axum_typed_multipart` with `utoipa`

The example is self-contained and I included a `Cargo.toml` because it requires additional dependencies. I haven't included a README as the code is hopefully reasonably easy to follow. On starting the web server a `rapidoc` user interface will describe the API endpoint and present a sample form that can be filled.

Thanks

murar8 reviewed (commented):
Hi @mattdavis90, thanks for the contribution! Maybe we could briefly mention this example in `src/lib.rs`? So users know it exists.

murar8 commented:
I also see the CI lint failing so I would suggest you install the pre-commit hooks to make your life easier.

mattdavis90 commented:
I went back and fixed the git commit message to be semantic and I've run through pre-commit - I totally missed that in the repo.

I've added documentation - hopefully it reads ok and is somewhat in keeping with what you already had. Thanks

murar8 commented on src/lib.rs:
````diff
@@ -251,6 +251,20 @@
+//! `axum_typed_multipart` which uses either `Bytes` or [tempfile::NamedTempFile](tempfile_3::NamedTempFile)
+//! as above. It is possible to get the best of both worlds as shown in the example.
+//! ```rust,no_run
+#![doc = include_str!("../examples/utoipa/src/main.rs")]
````
I would just provide a link here since it's a bit of a niche use case.

murar8 reviewed (approved):
Looks good, just left a small comment :)

mattdavis90 commented:
I've updated the docs to be a link and I _think_ I fixed the issues with the conflicting versions of `futures-core`

murar8 commented:
> I've updated the docs to be a link and I _think_ I fixed the issues with the conflicting versions of `futures-core`

If you want you could also put whatever extra dep in dev-dependencies in the root Cargo.toml instead of using an extra manifest, maybe it will fix the issue?

mattdavis90 commented:
I wasn't sure if it was reasonable to make devs install all dependencies for examples. Happy to go with your preference :)

murar8 commented:
Never mind @mattdavis90 the problem is independent of your PR, will get it fixed.
